p256-cm4/src/new_api.rs

```diff
@@ -0,0 +1,279 @@
+
+        let (x, y) = xy
+            .split_at_checked(32)
+            .ok_or(VerifyingKeySec1Error::TooLittleData)?;
+
+        let x: &[u8; 32] = x.try_into().unwrap();
+
+        if tag == 4 || tag == 6 || tag == 7 {
+            let y: &[u8; 32] = y
+                .try_into()
+                .map_err(|_| VerifyingKeySec1Error::TooLittleData)?;
+
+            if tag == 6 || tag == 7 {
+                let expected_parity = tag & 0x1;
+                let parity = y[31] & 0x1;
+
+                if expected_parity != parity {
+                    return Err(VerifyingKeySec1Error::InvalidParity);
+                }
+            }
+
+            Self::from_parts(x, y).map_err(VerifyingKeySec1Error::Key)
+        } else if tag == 2 || tag == 3 {
+            let x = to_little_endian(x);
+            let mut y = [0u32; 8];
+
+            let valid = sys::decompress_point(&mut y, &x, (u32::from(tag) & 0x1) == 1);
+
+            if valid {
+                Self::from_parts_valid_range(x, y).map_err(VerifyingKeySec1Error::Key)
+            } else {
+                Err(VerifyingKeySec1Error::Key(VerifyingKeyError::OutOfRange))
+            }
+        } else {
+            Err(VerifyingKeySec1Error::InvalidTag)
+        }
+    }
+
+    /// Convert this [`VerifyingKey`] into a tuple of little-endian integers
+    /// describing the `x` and `y` coordinates.
+    pub fn to_bytes(&self) -> ([u8; 32], [u8; 32]) {
+        let mut x = [0u8; 32];
+        let mut y = [0u8; 32];
+
+        to_big_endian(&self.x, &mut x);
+        to_big_endian(&self.y, &mut y);
+
+        (x, y)
+    }
+}
+
+/// A signature.
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, Clone, PartialEq)]
+pub struct Signature {
+    r: [u32; 8],
+    s: [u32; 8],
+}
+
+impl Signature {
+    /// Create a new [`Signature`] from the provided bytes.
+    ///
+    /// The first 32 bytes of the `bytes` are interpreted as the `r`
+    /// value of this [`Signature`], and the second 32 bytes are interpreted
+    /// as the `s` value. An error is returned if the two values don't
+    /// constitute a valid [`Signature`] (see [`Signature::from_parts`] for
+    /// more information on validity).
+    pub fn from_bytes(bytes: &[u8; 64]) -> Option<Self> {
+        Self::from_parts(
+            bytes[..32].try_into().unwrap(),
+            bytes[32..].try_into().unwrap(),
+        )
+    }
+
+    /// Create a new [`Signature`] from the little-endian encoded values
+    /// `r` and `s`.
+    ///
+    /// An error is returned if `r` or `s` is not in the range `1..=n - 1`, where
+    /// `n` is the `p256` order.
+    pub fn from_parts(r: &[u8; 32], s: &[u8; 32]) -> Option<Self> {
+        let r = to_little_endian(r);
+        let s = to_little_endian(s);
+
+        if !check_range_n(&r) || !check_range_n(&s) {
+            return None;
+        }
+
+        Some(Self { r, s })
+    }
+}
+
+/// A signing key, also called a private key.
+pub struct SigningKey {
```

Comment:
What do you think about renaming this to [SecretKey](https://docs.rs/p256/0.14.0-rc.0/p256/type.SecretKey.html) to match the `p256` crate naming?